test/integration/testRowMode.js

```diff
@@ -0,0 +1,188 @@
+    return function (done) {
+      const stmt = connection.execute({
+        sqlText: sql,
+        rowMode: testCase.rowMode,
+        streamResult: true
+      });
+      stmt.streamRows()
+        .on('data', function (row) {
+          assert.deepStrictEqual(row, testCase.expected)
+        })
+        .on('end', function () {
+          done();
+        })
+        .on('error', function (err) {
+          done(err)
+        });
+    }
+  };
+
+  const createItCallbackFetchResult = function (testCase) {
+    return function (done) {
+      connection.execute({
+        sqlText: sql,
+        rowMode: testCase.rowMode,
+        streamResult: false,
+        complete: function (err, stmt, rows) {
+          if (err) {
+            done(err);
+          }
+          assert.deepStrictEqual(rows[0], testCase.expected);
+          done();
+        }
+      })
+    }
+  };
+  describe('test incorect row mode - connection', function () {
+    it('test incorrect row mode', function (done) {
+      try {
+        connection = testUtil.createConnection({rowMode: 'invalid'});
+      } catch (err) {
+        assert.strictEqual(err.code, 411006);
+        done();
+      }
+    });
+  });
+  describe('test incorect row mode - statement', function () {
+    before(function (done) {
+      connection = testUtil.createConnection();
+      testUtil.connect(connection, done);
+    });
+
+    after(function (done) {
+      testUtil.destroyConnection(connection, done);
+    });
+
+    it('test incorrect row mode', function (done) {
+      try {
+        connection.execute({
+          rowMode: 'invalid',
+          sqlText: sql,
+          complete: function (err, stmt, rows) {
+          }
+        });
+      } catch (err) {
+        assert.strictEqual(err.code, 411006);
+        done();
+      }
+    });
+  });
+
+
+  describe('rowMode object in connection', function () {
+      before(function (done) {
+        connection = testUtil.createConnection({rowMode: RowMode.OBJECT});
+        testUtil.connect(connection, done);
+      });
+      after(function (done) {
+        testUtil.destroyConnection(connection, done);
+      });
+
+      const objectTestCases = testCases.filter(params => params.connectionRowMode === "object");
+
+      describe('stream', function () {
+        objectTestCases.forEach(testCase => it(testCase.name, createItCallbackStreamResult(testCase)));
+      })
+      describe('fetch', function () {
+        objectTestCases.forEach(testCase => it(testCase.name, createItCallbackFetchResult(testCase)));
+      })
+    }
+  );
+
+  describe('rowMode array in connection', function () {
+    before(function (done) {
+      connection = testUtil.createConnection({rowMode: RowMode.ARRAY});
+      testUtil.connect(connection, done);
+    });
+    after(function (done) {
+      testUtil.destroyConnection(connection, done);
+    });
+
+    const arrayTestCases = testCases.filter(params => params.connectionRowMode === "array");
```

Comment:
why do we need such configuration? why cannot we setup all parameters in testCases and don't exclude anything here?

Reply:
I've simplified test configuration